panoptes_client/project.py

```diff
@@ -197,6 +196,38 @@ def add_attached_image(
             }},
         )
 
+    def copy(self, new_subject_set_name=None):
+        """
+        Copy this project to a new project that will be owned by the
+        currently authenticated user.
+
+        A new_subject_set_name string argument can be passed which will be
+        used to name a new SubjectSet for the copied project.
+        This is useful for having an upload target straight after cloning.
+
+        Examples::
+
+            project.copy()
+            project.copy("My new subject set for uploading")
+        """
+        payload = {}
+        if new_subject_set_name:
+            payload['create_subject_set'] = new_subject_set_name
+
+        response = self.http_post(
+            '{}/copy'.format(self.id),
+            json=payload,
+        )
+
+        # find the API resource response in the response tuple
+        resource_response = response[0]
+        # extract the raw copied project resource response
+        raw_resource_response = resource_response[self._api_slug][0]
+        # convert it into a new project model representation
+        copied_project = Project(raw_resource_response)
```

Comment:
I suspect this may not behave exactly the way you want. The response probably won't have the correct etag, so the new project probably won't be saveable. That could be a problem since I imagine the first thing people will want to do when using this method is often going to be to modify and save the new project.

I suggest testing the above scenario, and if necessary call `copied_project.reload()` before returning it. I know that adds an extra API request every time, but I think it's worth it in this case.

Reply:
yep I've tested this and the newly minted resource wasn't savable due to the missing etag. 

FWIW I've re-used the etag from the copy response as it'll be the same as the round trip re GET response etag so this avoids the extra API call.